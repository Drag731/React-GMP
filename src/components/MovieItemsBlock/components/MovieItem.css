.b-item {
  max-width: 300px;
  margin-top: 30px; }
  .b-item__img-wrap {
    margin-bottom: 10px; }
  .b-item__img-pic {
    width: 100%; }
  .b-item__title-wrap {
    display: flex;
    justify-content: space-between;
    align-items: center; }
  .b-item__title-text {
    color: black;
    font-weight: bold; }
  .b-item__title-year {
    border: 1px solid black;
    border-radius: 5px;
    padding: 3px; }
  .b-item__ganre-wrap {
    color: darkgray; }
