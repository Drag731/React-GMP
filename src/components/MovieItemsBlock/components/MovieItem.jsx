import React from 'react';

const MovieItem = ({item, handleGoToDescriptionPage}) => (
    <div className="movie-items__item b-item">
        <div className="b-item__img-wrap">
            <img className="b-item__img-pic" src={item.poster_path} alt=""/>
        </div>
        <div className="b-item__title-wrap">
            <span className="b-item__title-text">{item.title}</span>
            <span className="b-item__title-year">{item.release_date}</span>
        </div>
        <div className="b-item__ganre-wrap">
            <span className="b-item__ganre-text">{item.genres.join(' & ')}</span>
        </div>
    </div>
);

export default MovieItem;