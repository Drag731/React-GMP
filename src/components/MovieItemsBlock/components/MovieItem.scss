.b-item {
    max-width: 300px;
    margin-top: 30px;
    &__img {
        &-wrap {
            margin-bottom: 10px;
        }
        &-pic {
            width: 100%;
        }
    }
    &__title {
        &-wrap {
            display: flex;
            justify-content: space-between;
            align-items: center;
        }
        &-text {
            color: black;
            font-weight: bold;
        }
        &-year {
            border: 1px solid black;
            border-radius: 5px;
            padding: 3px;
        }
    }
    &__ganre {
        &-wrap {
            color: darkgray;
        }
    }
}