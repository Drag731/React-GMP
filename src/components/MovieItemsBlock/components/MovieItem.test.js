import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import {shallow, mount, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import MovieItem from './MovieItem.jsx'

configure({ adapter: new Adapter() });

describe('<MovieItem  />', () => {
    it('Renders MovieItem component', () => {
        const item = {
            poster_path: 'http',
            id: 1,
            title: 'title',
            release_date: 2001,
            genres: ['Comedy']
        };

        const tree = shallow(
            <MovieItem item={item}/>
        );
        expect(tree.find('.movie-items__item')).toHaveLength(1);
    });
});