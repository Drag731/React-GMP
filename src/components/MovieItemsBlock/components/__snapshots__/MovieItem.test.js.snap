// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieItem  /> Snapshot one 1`] = `
<div
  className="movie-items__item b-item"
>
  <div
    className="b-item__img-wrap"
  >
    <img
      alt=""
      className="b-item__img-pic"
      src="http"
    />
  </div>
  <div
    className="b-item__title-wrap"
  >
    <a
      className="b-item__title-link"
      href="/filmzilla/description/1"
      onClick={[Function]}
    >
      <span
        className="b-item__title-text"
      >
        title
      </span>
    </a>
    <span
      className="b-item__title-year"
    >
      2001
    </span>
  </div>
  <div
    className="b-item__ganre-wrap"
  >
    <span
      className="b-item__ganre-text"
    >
      Comedy
    </span>
  </div>
</div>
`;
